homeassistant/components/samsungtv/media_player.py

```diff
@@ -348,17 +314,15 @@ def media_previous_track(self):
             await asyncio.sleep(KEY_PRESS_TIMEOUT, self.hass.loop)
         await self.hass.async_add_job(self.send_key, "KEY_ENTER")
 
-    def turn_on(self):
+    async def async_turn_on(self):
         """Turn the media player on."""
-        if self._mac:
-            wakeonlan.send_magic_packet(self._mac, ip_address=self._broadcast)
-        else:
-            self.send_key("KEY_POWERON")
+        if self._on_script:
+            await self._on_script.async_run()
 
-    async def async_select_source(self, source):
+    def select_source(self, source):
         """Select input source."""
         if source not in SOURCES:
             LOGGER.error("Unsupported source")
             return
 
-        await self.hass.async_add_job(self.send_key, SOURCES[source])
+        self.hass.async_add_job(self.send_key, SOURCES[source])
```

Comment:
Since we changed the method definition above, we should just call `send_key` directly, and not add a job. Or revert the definition change.